server/leafnode.go

```diff
@@ -1271,16 +1285,25 @@ func (c *client) processLeafSub(argo []byte) (err error) {
 
 	acc := c.acc
 	// Check if we have a loop.
-	if strings.HasPrefix(string(sub.subject), leafNodeLoopDetectionSubjectPrefix) && string(sub.subject) == acc.getLDSubject() {
+	ldsPrefix := bytes.HasPrefix(sub.subject, []byte(leafNodeLoopDetectionSubjectPrefix))
+	if ldsPrefix && string(sub.subject) == acc.getLDSubject() {
 		c.mu.Unlock()
 		srv.reportLeafNodeLoop(c)
 		return nil
 	}
 
-	// Check permissions if applicable.
-	if !c.canExport(string(sub.subject)) {
+	// Check permissions if applicable. (but exclude the $LDS, $GR and _GR_)
+	checkPerms := true
+	if sub.subject[0] == '$' || sub.subject[0] == '_' {
```

Comment:
Even though it's not super explicit, it's my understanding that splitArg wouldn't return len(sub.subject) == 0.

Reply:
We would return error above in this function if we don't have a subject, so that is safe.